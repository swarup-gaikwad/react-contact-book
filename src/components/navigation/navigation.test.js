import React from "react";
import { shallow } from "enzyme";
import Navigation from "./navigation";

describe("Navigation", () => {
  it("should render correctly", () => {
    const component = shallow(<Navigation />);
  });
  it("should render initial layout", () => {
    // when
    const component = shallow(<Navigation />);
    // then
    expect(component.getElements()).toMatchSnapshot();
  });
});
