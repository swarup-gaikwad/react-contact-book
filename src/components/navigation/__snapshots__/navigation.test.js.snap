// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render initial layout 1`] = `
Array [
  <BrowserRouter>
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <div>
              <div
                className="d-inline-flex"
              >
                <Link
                  className="nav-item nav-link"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
                <Link
                  className="nav-item nav-link"
                  replace={false}
                  to="/add-contact"
                >
                  Add Contact
                </Link>
                <Link
                  className="nav-item nav-link"
                  replace={false}
                  to="/contacts"
                >
                  All Contacts
                </Link>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/contacts"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/add-contact"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/edit-contact"
      />
    </div>
  </BrowserRouter>,
]
`;
